pkg/utils/secrets/basic_auth.go

```diff
@@ -43,7 +35,9 @@ const (
 
 // BasicAuthSecretConfig contains the specification for a to-be-generated basic authentication secret.
 type BasicAuthSecretConfig struct {
-	Name   string
+	Name string
+	// Do not remove the 'formatType' field, even though only one format 'BasicAuthFormatNormal' is supported,
+	// because it will trigger recreation of all Basic Auth secrets and may affect observability users credentials.
```

Comment:
```suggestion
	// Format is the format type.
	//
	// Do not remove this field, even though the field is not used and there is only one supported format ("normal").
	// The secret manager computes the Secret hash based on the config object (BasicAuthSecretConfig). A field removal in the
	// BasicAuthSecretConfig object would compute a new Secret hash and this would lead the existing Secrets to be regenerated.
	// Hence, usages of the BasicAuthSecretConfig should continue to pass the Format field with value "normal".
```